core/types/calls.cc

```diff
@@ -995,12 +1002,42 @@ DispatchResult dispatchCallSymbol(const GlobalState &gs, const DispatchArgs &arg
                     continue;
                 }
                 consumed.insert(spec.name);
-                TypeAndOrigins tpe;
-                tpe.origins = {kwargsLoc};
+                // We want the precise location of the kwarg that we are processing.
+                // To do that, we rely on hash->{keys,values} being added in the order
+                // that the kwargs appear in the the call.  We also rely on args from
+                // a shapely-typed kwsplat hash (implicit or explicit) being added to
+                // hash->{keys,values} after the explicitly-named kwargs in the call.
                 auto offset = arg - hash->keys.begin();
+                // offset * 2 skips kwargs we have already seen.
+                // + 1 moves us to the value being passed.
+                ENFORCE(!sawKwSplat);
+                auto kwaValue = kwait + (offset * 2) + 1;
+                TypeAndOrigins tpe;
+                ENFORCE(args.args.size() == args.locs.args.size(), "what?!");
```

Comment:
```suggestion
                ENFORCE(args.args.size() == args.locs.args.size(), "Send arg and loc vectors did not match in length");
```

Reply:
Whoops, sorry, I should have made this a better message.